diff --git a/algo/kernels/basic/hifi4/xa_nn_reduce_asym16s_asym16s.c b/algo/kernels/basic/hifi4/xa_nn_reduce_asym16s_asym16s.c
index 44931fc..7c5dc5c 100644
--- a/algo/kernels/basic/hifi4/xa_nn_reduce_asym16s_asym16s.c
+++ b/algo/kernels/basic/hifi4/xa_nn_reduce_asym16s_asym16s.c
@@ -940,7 +940,7 @@ WORD32 xa_nn_reduce_mean_4D_asym16s_asym16s(WORD16 * __restrict__ p_out
   {
     current = p_axis[axis_itr];
     XA_NNLIB_ARG_CHK_COND(((current < 0) || (current > (num_inp_dims - 1))), -1);
-    XA_NNLIB_ARG_CHK_COND((p_inp_shape[current] > 1024), -1);
+    // XA_NNLIB_ARG_CHK_COND((p_inp_shape[current] > 1024), -1);
 
     /* Avoid calculation in case of repeated axis dims*/
     if(current != past)
diff --git a/algo/kernels/cnn/hifi4/xa_nn_conv2d_std_sym8sxsym16s.c b/algo/kernels/cnn/hifi4/xa_nn_conv2d_std_sym8sxsym16s.c
index bc1078e..28abf4c 100644
--- a/algo/kernels/cnn/hifi4/xa_nn_conv2d_std_sym8sxsym16s.c
+++ b/algo/kernels/cnn/hifi4/xa_nn_conv2d_std_sym8sxsym16s.c
@@ -778,7 +778,8 @@ WORD32 xa_nn_dilated_conv2d_std_v2_per_chan_sym8sxsym16s(
   WORD32 out_width_over_x_r_pad = 0;
   // Determine x-right padding
   WORD32 x_r_pad = kernel_width_dilation + (out_width - 1) * x_stride - (x_padding + input_width);//dilation
-  XA_NNLIB_ARG_CHK_COND((x_r_pad<0), -1);
+  x_r_pad = x_r_pad < 0 ? 0 : x_r_pad;
+  // XA_NNLIB_ARG_CHK_COND((x_r_pad<0), -1);
   if(x_r_pad >= kernel_width_dilation)//dilation
   {
     out_width_over_x_r_pad = conv_x_right_pad(x_padding, input_width, x_stride, out_width, out_height, out_channels, out_channels_offset, out_width_offset, out_height_offset, p_bias, p_out, p_out_multiplier, p_out_shift, out_activation_min, out_activation_max);
@@ -787,7 +788,8 @@ WORD32 xa_nn_dilated_conv2d_std_v2_per_chan_sym8sxsym16s(
 
   // Determine y-bottom padding
   WORD32 y_b_pad = kernel_height_dilation + (out_height - 1) * y_stride - (y_padding + input_height);
-  XA_NNLIB_ARG_CHK_COND((y_b_pad<0), -1);
+  y_b_pad = y_b_pad < 0 ? 0 : y_b_pad;
+  // XA_NNLIB_ARG_CHK_COND((y_b_pad<0), -1);
 
   XA_NNLIB_ARG_CHK_COND((kernel_height_dilation > ( y_padding + input_height + y_b_pad)), -1);//dilation
   XA_NNLIB_ARG_CHK_COND((kernel_width_dilation  > ( x_padding + input_width  + x_r_pad)), -1);//dilation
diff --git a/algo/kernels/basic/hifi4/xa_nn_elm_quantize.c b/algo/kernels/basic/hifi4/xa_nn_elm_quantize.c
index df5ba5a..0a9fb28 100644
--- a/algo/kernels/basic/hifi4/xa_nn_elm_quantize.c
+++ b/algo/kernels/basic/hifi4/xa_nn_elm_quantize.c
@@ -2012,8 +2012,6 @@ WORD32 xa_nn_elm_quantize_f32_asym16s(WORD16 * __restrict__ p_out,
   ae_int32x2 quant_max = AE_MOVDA32(32767);
   ae_int32x2 quant_min = AE_MOVDA32(-32768);
   xtfloatx2 d_out_scale = (xtfloatx2)*out_scale_ptr;
-  xtfloatx2 d_one = XT_FLOAT_SX2(AE_MOVDA32(1), 0);
-  xtfloatx2 d_one_over_out_scale = XT_DIV_SX2(d_one, d_out_scale);
 
   for(i = 0; i < (num_elm >> 2); i++)
   {
@@ -2023,8 +2021,8 @@ WORD32 xa_nn_elm_quantize_f32_asym16s(WORD16 * __restrict__ p_out,
 
     XT_LASX2IP(d_inp0, align_inp, p_i);
     XT_LASX2IP(d_inp1, align_inp, p_i);
-    d_inp0_t = XT_MUL_SX2(d_inp0, d_one_over_out_scale);
-    d_inp1_t = XT_MUL_SX2(d_inp1, d_one_over_out_scale);
+    d_inp0_t = XT_DIV_SX2(d_inp0, d_out_scale);
+    d_inp1_t = XT_DIV_SX2(d_inp1, d_out_scale);
     d_inp0_t = XT_FIROUND_SX2(d_inp0_t);
     d_inp1_t = XT_FIROUND_SX2(d_inp1_t);    
     d_out32_0 = XT_TRUNC_SX2(d_inp0_t, 0);
@@ -2045,7 +2043,7 @@ WORD32 xa_nn_elm_quantize_f32_asym16s(WORD16 * __restrict__ p_out,
     xtfloat d_inp0_t;
     ae_int32x2 d_out32_0;
     XT_LSIP(d_inp0, (xtfloat *)p_i, sizeof(FLOAT32));
-    d_inp0_t = XT_MUL_S(d_inp0, d_one_over_out_scale);
+    d_inp0_t = XT_DIV_S(d_inp0, d_out_scale);
     d_inp0_t = XT_FIROUND_S(d_inp0_t);
     d_out32_0 = XT_TRUNC_S(d_inp0_t, 0);    
     d_out32_0 = AE_ADD32S(d_out32_0, d_out_zero_bias);
