diff --git a/algo/kernels/basic/hifi_iq/xa_nn_elm_logical_bool.c b/algo/kernels/basic/hifi_iq/xa_nn_elm_logical_bool.c
index 9e40908..18be8fe 100644
--- a/algo/kernels/basic/hifi_iq/xa_nn_elm_logical_bool.c
+++ b/algo/kernels/basic/hifi_iq/xa_nn_elm_logical_bool.c
@@ -126,8 +126,8 @@ WORD32 xa_nn_elm_logicaland_boolxbool_bool(WORD8 * __restrict__ p_out,
     // remainder loop
     if(rem_length)
     {   
-        AE_LA2NX8_XP(m1, align_dst, p_in1, rem_length);
-        AE_LA2NX8_XP(m2, align_dst, p_in2, rem_length);
+        AE_LA2NX8_XP(m1, align_src_in1, p_in1, rem_length);
+        AE_LA2NX8_XP(m2, align_src_in2, p_in2, rem_length);
 
         m1 = AE_OR2NX8(m1, m2);
         AE_SAV2NX8_XP(m1, align_dst, p_o, rem_length);
diff --git a/algo/kernels/pool/hifi_iq/xa_nn_maxpool_16_nhwc.c b/algo/kernels/pool/hifi_iq/xa_nn_maxpool_16_nhwc.c
index 43593cc..6623eba 100644
--- a/algo/kernels/pool/hifi_iq/xa_nn_maxpool_16_nhwc.c
+++ b/algo/kernels/pool/hifi_iq/xa_nn_maxpool_16_nhwc.c
@@ -170,6 +170,7 @@ const WORD16* __restrict__ p_inp,
                     xb_vecNx16 v;
                     valign va;
                     for(i = 0; i < 16; i++) tmp[i] = p_s[i * x_stride];
+                    p_tmp = (xb_vecNx16 *)tmp;
                     va = AE_LA_PP((void *)p_tmp);
                     AE_LANX16_IP(v, va, p_tmp);
                     m = AE_MAXNX16(m, v);
@@ -187,7 +188,8 @@ const WORD16* __restrict__ p_inp,
                         xb_vecNx16 v;
                         valign va;
                         for(i = 0; i < rem_w; i++) tmp[i] = p_s[i * x_stride];
-                        va = AE_LA_PP((void *)tmp);
+                        p_tmp = (xb_vecNx16 *)tmp;
+                        va = AE_LA_PP((void *)p_tmp);
                         AE_LAVNX16_XP(v, va, p_tmp, rem_w << 1);
                         m = AE_MAXNX16(m, v);
                     }
