diff --git a/algo/kernels/basic/hifi5/xa_nn_reduce_asym16s_asym16s.c b/algo/kernels/basic/hifi5/xa_nn_reduce_asym16s_asym16s.c
index cd495e1..5901f56 100644
--- a/algo/kernels/basic/hifi5/xa_nn_reduce_asym16s_asym16s.c
+++ b/algo/kernels/basic/hifi5/xa_nn_reduce_asym16s_asym16s.c
@@ -1205,7 +1205,7 @@ WORD32 xa_nn_reduce_mean_4D_asym16s_asym16s(WORD16 * __restrict__ p_out
   {
     current = p_axis[axis_itr];
     XA_NNLIB_ARG_CHK_COND(((current < 0) || (current > (num_inp_dims - 1))), -1);
-    XA_NNLIB_ARG_CHK_COND((p_inp_shape[current] > 1024), -1);
+    // XA_NNLIB_ARG_CHK_COND((p_inp_shape[current] > 1024), -1);
 
     /* Avoid calculation in case of repeated axis dims*/
     if(current != past)
diff --git a/algo/kernels/cnn/hifi5/xa_nn_conv2d_std_sym8sxsym16s.c b/algo/kernels/cnn/hifi5/xa_nn_conv2d_std_sym8sxsym16s.c
index ea2e6f4..585c58f 100644
--- a/algo/kernels/cnn/hifi5/xa_nn_conv2d_std_sym8sxsym16s.c
+++ b/algo/kernels/cnn/hifi5/xa_nn_conv2d_std_sym8sxsym16s.c
@@ -637,8 +637,8 @@ WORD32 xa_nn_dilated_conv2d_std_v2_per_chan_sym8sxsym16s(
   WORD32 out_width_over_x_r_pad = 0;
   // Determine x-right padding
   WORD32 x_r_pad = kernel_width_dilation + (out_width - 1) * x_stride - (x_padding + input_width);//dilation
-  //x_r_pad = x_r_pad < 0 ? 0 : x_r_pad;
-  XA_NNLIB_ARG_CHK_COND((x_r_pad<0), -1);
+  x_r_pad = x_r_pad < 0 ? 0 : x_r_pad;
+  // XA_NNLIB_ARG_CHK_COND((x_r_pad<0), -1);
   if(x_r_pad >= kernel_width_dilation)//dilation
   {
     out_width_over_x_r_pad = conv_x_right_pad(x_padding, input_width, x_stride, out_width, out_height, out_channels, out_channels_offset, out_width_offset, out_height_offset, p_bias, p_out, p_out_multiplier, p_out_shift, out_zero_bias, out_activation_min, out_activation_max);
@@ -647,8 +647,8 @@ WORD32 xa_nn_dilated_conv2d_std_v2_per_chan_sym8sxsym16s(
 
   // Determine y-bottom padding
   WORD32 y_b_pad = kernel_height_dilation + (out_height - 1) * y_stride - (y_padding + input_height);
-  //y_b_pad = y_b_pad < 0 ? 0 : y_b_pad;
-  XA_NNLIB_ARG_CHK_COND((y_b_pad<0), -1);
+  y_b_pad = y_b_pad < 0 ? 0 : y_b_pad;
+  // XA_NNLIB_ARG_CHK_COND((y_b_pad<0), -1);
 
 
   XA_NNLIB_ARG_CHK_COND((kernel_height_dilation > ( y_padding + input_height + y_b_pad)), -1);//dilation
diff --git a/algo/kernels/basic/hifi5/xa_nn_elm_quantize.c b/algo/kernels/basic/hifi5/xa_nn_elm_quantize.c
index a658d11..531cc43 100644
--- a/algo/kernels/basic/hifi5/xa_nn_elm_quantize.c
+++ b/algo/kernels/basic/hifi5/xa_nn_elm_quantize.c
@@ -1584,23 +1584,35 @@ WORD32 xa_nn_elm_quantize_f32_asym16s(WORD16 * __restrict__ p_out,
   ae_int32x2 d_out_zero_bias = SW_MOVDA32(out_zero_bias);
   xtfloat *out_scale_ptr = (xtfloat*)&out_scale;
   xtfloatx2 d_out_scale = AE_MOVXTFLOATX2_FROMXTFLOAT(*out_scale_ptr);
-  xtfloatx2 d_one = FLOAT_SX2(SW_MOVDA32(1),0);
-  xtfloatx2 d_one_over_out_scale = XT_DIV_SX2(d_one, d_out_scale);
+#ifdef DIV_SX4
+  xtfloatx4 d_out_scalex4 = xtfloat_rtor_xtfloatx4(*out_scale_ptr);
+#endif
 
   for(i = 0; i < (num_elm >> 3); i++)
   {
-    xtfloatx2 d_inp0, d_inp1, d_inp2, d_inp3;
     xtfloatx2 d_inp0_t, d_inp1_t, d_inp2_t, d_inp3_t;
     ae_int16x4 d_out0, d_out1;
     ae_int32x2 d_out32_0, d_out32_1, d_out32_2, d_out32_3;
 
-
+#ifdef DIV_SX4
+    xtfloatx4 d_inp01, d_inp23;
+    AE_LASX4IP(d_inp01, align_inp, p_i);
+    AE_LASX4IP(d_inp23, align_inp, p_i);
+    d_inp01 = DIV_SX4(d_inp01, d_out_scalex4);
+    d_inp23 = DIV_SX4(d_inp23, d_out_scalex4);
+    d_inp0_t = AE_EXTRACTSX2_FROMSX4_H(d_inp01);
+    d_inp1_t = AE_EXTRACTSX2_FROMSX4_L(d_inp01);
+    d_inp2_t = AE_EXTRACTSX2_FROMSX4_H(d_inp23);
+    d_inp3_t = AE_EXTRACTSX2_FROMSX4_L(d_inp23);
+#else
+    xtfloatx2 d_inp0, d_inp1, d_inp2, d_inp3;
     AE_LASX2X2_IP(d_inp0, d_inp1, align_inp, p_i);
     AE_LASX2X2_IP(d_inp2, d_inp3, align_inp, p_i);
-
-    MUL_SX2X2(d_inp0_t, d_inp1_t, d_inp0, d_inp1, d_one_over_out_scale, d_one_over_out_scale);
-    MUL_SX2X2(d_inp2_t, d_inp3_t, d_inp2, d_inp3, d_one_over_out_scale, d_one_over_out_scale);
-
+    d_inp0_t = DIV_SX2(d_inp0, d_out_scale);
+    d_inp1_t = DIV_SX2(d_inp1, d_out_scale);
+    d_inp2_t = DIV_SX2(d_inp2, d_out_scale);
+    d_inp3_t = DIV_SX2(d_inp3, d_out_scale);
+#endif
     d_inp0_t = XT_FIROUND_SX2(d_inp0_t);
     d_inp1_t = XT_FIROUND_SX2(d_inp1_t);
     d_inp2_t = XT_FIROUND_SX2(d_inp2_t);
@@ -1649,8 +1661,10 @@ WORD32 xa_nn_elm_quantize_f32_asym16s(WORD16 * __restrict__ p_out,
     d_inp2 = AE_MOVXTFLOATX2_FROMINT32X2(AE_MOVINT32X2_FROMINT16X4(d_tmp2));
     d_inp3 = AE_MOVXTFLOATX2_FROMINT32X2(AE_MOVINT32X2_FROMINT16X4(d_tmp3));
 
-    MUL_SX2X2(d_inp0_t, d_inp1_t, d_inp0, d_inp1, d_one_over_out_scale, d_one_over_out_scale);
-    MUL_SX2X2(d_inp2_t, d_inp3_t, d_inp2, d_inp3, d_one_over_out_scale, d_one_over_out_scale);
+    d_inp0_t = DIV_SX2(d_inp0, d_out_scale);
+    d_inp1_t = DIV_SX2(d_inp1, d_out_scale);
+    d_inp2_t = DIV_SX2(d_inp2, d_out_scale);
+    d_inp3_t = DIV_SX2(d_inp3, d_out_scale);
 
     d_inp0_t = XT_FIROUND_SX2(d_inp0_t);
     d_inp1_t = XT_FIROUND_SX2(d_inp1_t);
diff --git a/algo/kernels/activations/hifi5/xa_nn_softmax_asym8_asym8.c b/algo/kernels/activations/hifi5/xa_nn_softmax_asym8_asym8.c
index 9599fd9..3902ca7 100644
--- a/algo/kernels/activations/hifi5/xa_nn_softmax_asym8_asym8.c
+++ b/algo/kernels/activations/hifi5/xa_nn_softmax_asym8_asym8.c
@@ -591,7 +591,7 @@ WORD32 xa_nn_vec_softmax_asym8s_asym8s( WORD8 * __restrict__ p_out,
     xtbool4 g3210;
     WORD8 *p_in = (WORD8 *)p_vec;
     ae_int32x4 *p_exp = (ae_int32x4 *)ALIGN_PTR(p_scratch, ALIGNMENT);
-    ae_int32x2 x32, x10/*, diff_min*/;
+    ae_int32x2 x32, x10, diff_min;
     ae_int32x2 dequantized_x32, dequantized_x10;
     ae_f32x2 unsat_out76, unsat_out54, unsat_out32, unsat_out10;
     ae_int32x2 ONE;
@@ -668,8 +668,8 @@ WORD32 xa_nn_vec_softmax_asym8s_asym8s( WORD8 * __restrict__ p_out,
         }
     }
 
-//    diff_min = AE_MOVDA32(diffmin);
-    ae_int16x4 diff_min16 = AE_MOVDA16(diffmin);
+    diff_min = AE_MOVDA32(diffmin);
+    ae_int16x4 diff_min16 = AE_SAT16X4(diff_min, diff_min);
     sum_exp = AE_MOVF32X2_FROMINT32X2(z); // setting to zero
 
     p_in = (WORD8 *)p_vec;
