diff --git a/algo/kernels/basic/hifi5/xa_nn_reduce_asym16s_asym16s.c b/algo/kernels/basic/hifi5/xa_nn_reduce_asym16s_asym16s.c
index cd495e1..5901f56 100644
--- a/algo/kernels/basic/hifi5/xa_nn_reduce_asym16s_asym16s.c
+++ b/algo/kernels/basic/hifi5/xa_nn_reduce_asym16s_asym16s.c
@@ -1205,7 +1205,7 @@ WORD32 xa_nn_reduce_mean_4D_asym16s_asym16s(WORD16 * __restrict__ p_out
   {
     current = p_axis[axis_itr];
     XA_NNLIB_ARG_CHK_COND(((current < 0) || (current > (num_inp_dims - 1))), -1);
-    XA_NNLIB_ARG_CHK_COND((p_inp_shape[current] > 1024), -1);
+    // XA_NNLIB_ARG_CHK_COND((p_inp_shape[current] > 1024), -1);
 
     /* Avoid calculation in case of repeated axis dims*/
     if(current != past)
diff --git a/algo/kernels/cnn/hifi5/xa_nn_conv2d_std_sym8sxsym16s.c b/algo/kernels/cnn/hifi5/xa_nn_conv2d_std_sym8sxsym16s.c
index ea2e6f4..585c58f 100644
--- a/algo/kernels/cnn/hifi5/xa_nn_conv2d_std_sym8sxsym16s.c
+++ b/algo/kernels/cnn/hifi5/xa_nn_conv2d_std_sym8sxsym16s.c
@@ -637,8 +637,8 @@ WORD32 xa_nn_dilated_conv2d_std_v2_per_chan_sym8sxsym16s(
   WORD32 out_width_over_x_r_pad = 0;
   // Determine x-right padding
   WORD32 x_r_pad = kernel_width_dilation + (out_width - 1) * x_stride - (x_padding + input_width);//dilation
-  //x_r_pad = x_r_pad < 0 ? 0 : x_r_pad;
-  XA_NNLIB_ARG_CHK_COND((x_r_pad<0), -1);
+  x_r_pad = x_r_pad < 0 ? 0 : x_r_pad;
+  // XA_NNLIB_ARG_CHK_COND((x_r_pad<0), -1);
   if(x_r_pad >= kernel_width_dilation)//dilation
   {
     out_width_over_x_r_pad = conv_x_right_pad(x_padding, input_width, x_stride, out_width, out_height, out_channels, out_channels_offset, out_width_offset, out_height_offset, p_bias, p_out, p_out_multiplier, p_out_shift, out_zero_bias, out_activation_min, out_activation_max);
@@ -647,8 +647,8 @@ WORD32 xa_nn_dilated_conv2d_std_v2_per_chan_sym8sxsym16s(
 
   // Determine y-bottom padding
   WORD32 y_b_pad = kernel_height_dilation + (out_height - 1) * y_stride - (y_padding + input_height);
-  //y_b_pad = y_b_pad < 0 ? 0 : y_b_pad;
-  XA_NNLIB_ARG_CHK_COND((y_b_pad<0), -1);
+  y_b_pad = y_b_pad < 0 ? 0 : y_b_pad;
+  // XA_NNLIB_ARG_CHK_COND((y_b_pad<0), -1);
 
 
   XA_NNLIB_ARG_CHK_COND((kernel_height_dilation > ( y_padding + input_height + y_b_pad)), -1);//dilation
